react/react1/week3/todo-tracker/src/App.js

```diff
@@ -0,0 +1,122 @@
+    )
+      .then((res) => res.json())
+      .then(
+        (data) => {
+          setArrLength(data.length);
+          data.forEach((e) => (e.taskCompletionStatus = "false"));
+          setFilterToDoList(data);
+          setFullToDoList(data);
+        },
+        (error) => {
+          console.log(error);
+        }
+      );
+  }, []);
+
+  return (
+    <div className="App">
+      <WatchCount />
+
+      <Search
+        onFilteredToDoListChange={(text) => {
+          if (text) {
+            setFilterToDoList(
+              filterToDoList.filter((tododata) =>
+                tododata.description.toLowerCase().includes(text.toLowerCase())
+              )
+            );
+          } else {
+            setFilterToDoList(fullToDoList);
+          }
+        }}
+      />
+
+      <AddToDo
+        onAddTodoList={(task, time) => {
+          const newId = parseInt(fullToDoList.length) + arrLength;
+          const newObj = {
+            id: "",
+            description: "",
+            deadline: "",
+            taskCompletionStatus: "",
+          };
+          newObj.id = (newId + 1).toString();
+          newObj.description = task.value;
+          newObj.deadline = time.value;
+          newObj.taskCompletionStatus = "false";
+          setFullToDoList([...fullToDoList, newObj]);
+          setFilterToDoList([...fullToDoList, newObj]);
+        }}
+      />
+      <ToDoList
+        ds={filterToDoList}
+        onChangeCheckBox={(text) => {
+          fullToDoList.forEach((element) => {
+            if (parseInt(element.id) === parseInt(text)) {
+              element.taskCompletionStatus =
+                element.taskCompletionStatus === "true" ? "false" : "true";
+            }
+          });
+          setFilterToDoList([...fullToDoList]);
+          setFullToDoList([...fullToDoList]);
+        }}
+        onDeleteItem={(text) => {
+          let newFullToDoList = fullToDoList.filter(
+            (element) => element.id !== text
+          );
+          setFilterToDoList([...newFullToDoList]);
+          setFullToDoList([...newFullToDoList]);
+        }}
+        onUpdateItem={(text) => {
+          if (document.getElementById(text + "button").innerText === "Edit") {
+            fullToDoList.forEach((element) => {
+              if (parseInt(element.id) === parseInt(text)) {
+                document.getElementById(text + "button").innerText = "Upload";
+                document.getElementById(element.id + "description").innerHTML =
+                  "<input type=text id=" + element.id + "task></input>";
+                document.getElementById(element.id + "deadline").innerHTML =
+                  "<input type='date' id=" + element.id + "date></input>";
+              }
+            });
+            setFilterToDoList([...fullToDoList]);
+            setFullToDoList([...fullToDoList]);
+          } else if (
+            document.getElementById(text + "button").innerText === "Upload"
```

Comment:
In React, or other frameworks, if you want to use a button, you create a button component with props and use it here. 
document.getElementById is a bit imperative. React is declaraive. There are ways to get the component in React if you need it, by using ref, but it is not necessary for this task.